diff --git a/node_modules/expo-updates/expo-updates-gradle-plugin/build.gradle.kts b/node_modules/expo-updates/expo-updates-gradle-plugin/build.gradle.kts
index 52cb353..2ad21b6 100644
--- a/node_modules/expo-updates/expo-updates-gradle-plugin/build.gradle.kts
+++ b/node_modules/expo-updates/expo-updates-gradle-plugin/build.gradle.kts
@@ -1,7 +1,7 @@
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 
 plugins {
-  kotlin("jvm") version("2.0.21")
+  kotlin("jvm") version("2.2.10")
   id("java-gradle-plugin")
 }
 
@@ -12,7 +12,7 @@ repositories {
 
 dependencies {
   implementation(gradleApi())
-  compileOnly("com.android.tools.build:gradle:8.5.0")
+  compileOnly("com.android.tools.build:gradle:9.0.0")
   implementation("com.facebook.react:react-native-gradle-plugin")
 }
 
@@ -22,8 +22,8 @@ java {
 }
 
 tasks.withType<KotlinCompile> {
-  kotlinOptions {
-    jvmTarget = JavaVersion.VERSION_11.toString()
+  compilerOptions {
+    jvmTarget.set(org.jetbrains.kotlin.gradle.dsl.JvmTarget.JVM_11)
   }
 }
 
