diff --git a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-plugin-shared/src/main/kotlin/expo/modules/plugin/ExpoGradleExtension.kt b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-plugin-shared/src/main/kotlin/expo/modules/plugin/ExpoGradleExtension.kt
index f79adf8..002fcba 100644
--- a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-plugin-shared/src/main/kotlin/expo/modules/plugin/ExpoGradleExtension.kt
+++ b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-plugin-shared/src/main/kotlin/expo/modules/plugin/ExpoGradleExtension.kt
@@ -12,7 +12,7 @@ data class AutolinkingOptions(
   val exclude: List<String>? = null
 ) {
   fun toJson(): String {
-    return Json.encodeToString(AutolinkingOptions.serializer(), this)
+    return Json.encodeToString(serializer(), this)
   }
 
   companion object {
diff --git a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-plugin/src/main/kotlin/expo/modules/plugin/ExpoAutolinkingPlugin.kt b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-plugin/src/main/kotlin/expo/modules/plugin/ExpoAutolinkingPlugin.kt
index 7a6f1bb..fea8adf 100644
--- a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-plugin/src/main/kotlin/expo/modules/plugin/ExpoAutolinkingPlugin.kt
+++ b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-plugin/src/main/kotlin/expo/modules/plugin/ExpoAutolinkingPlugin.kt
@@ -67,7 +67,7 @@ open class ExpoAutolinkingPlugin : Plugin<Project> {
         .sourceSets
         .getByName("main")
         .java
-        .srcDir(getPackageListDir(project))
+        .srcDir(getPackageListDir(project).get().asFile)
     }
   }
 
diff --git a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/SettingsManager.kt b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/SettingsManager.kt
index 85bcd5c..585b586 100644
--- a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/SettingsManager.kt
+++ b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/SettingsManager.kt
@@ -21,7 +21,6 @@ import groovy.lang.GroovyShell
 import org.gradle.api.Project
 import org.gradle.api.initialization.Settings
 import org.gradle.api.logging.Logging
-import org.gradle.internal.extensions.core.extra
 import java.io.File
 
 class SettingsManager(
@@ -109,7 +108,7 @@ class SettingsManager(
 
     // Defines the required features for the core module
     settings.gradle.beforeProject("expo-modules-core") { project ->
-      project.extra.set("coreFeatures", config.coreFeatures)
+      project.extensions.extraProperties.set("coreFeatures", config.coreFeatures)
     }
 
     settings.gradle.beforeRootProject { rootProject: Project ->
@@ -156,7 +155,7 @@ class SettingsManager(
         }
     }
 
-    settings.gradle.extensions.create("expoGradle", ExpoGradleExtension::class.java, config, autolinkingOptions, projectRoot)
+    settings.extensions.create("expoGradle", ExpoGradleExtension::class.java, config, autolinkingOptions, projectRoot)
   }
 
   /**
